incubator/catalog-schema-registry/src/main/java/io/aklivity/zilla/runtime/catalog/schema/registry/internal/config/SchemaRegistryOptionsConfigBuilder.java

```diff
@@ -53,9 +57,20 @@ public SchemaRegistryOptionsConfigBuilder<T> context(
         return this;
     }
 
+    public SchemaRegistryOptionsConfigBuilder<T> maxAge(
+        Duration maxAge)
+    {
+        this.maxAge = maxAge;
+        return this;
+    }
+
     @Override
     public T build()
     {
-        return mapper.apply(new SchemaRegistryOptionsConfig(url, context));
+        if (maxAge == null)
+        {
+            this.maxAge = Duration.ofSeconds(MAX_AGE_DEFAULT);
+        }
```

Comment:
Defaulting should not mutate the builder state, this should be a local variable instead, initialized to the field, tested for null and defaulted if needed, then passing the local variable to the constructor instead of the field.